crates/wdk-build/src/utils.rs

```diff
@@ -374,39 +398,154 @@ fn read_registry_key_string_value(
-            let major = major_str.parse::<u32>().ok()?;
-            let minor = minor_str.parse::<u32>().ok()?;
-            Some((major, minor))
-        })
+        .filter_map(|entry| BasicVersion::parse(entry.file_name().to_str()?))
         .max()
 }
 
 #[cfg(test)]
 mod tests {
+    use assert_fs::prelude::*;
+
     use super::*;
 
+    mod basic_version {
+        use super::*;
+
+        #[test]
+        fn test_new() {
+            let version = BasicVersion::new(1, 2);
+            assert_eq!(version.major, 1);
+            assert_eq!(version.minor, 2);
+        }
+
+        #[test]
+        fn test_parse_valid_versions() {
+            assert_eq!(BasicVersion::parse("1.2"), Some(BasicVersion::new(1, 2)));
+            assert_eq!(BasicVersion::parse("0.0"), Some(BasicVersion::new(0, 0)));
+            assert_eq!(
+                BasicVersion::parse("10.15"),
+                Some(BasicVersion::new(10, 15))
+            );
+            assert_eq!(
+                BasicVersion::parse("999.1"),
+                Some(BasicVersion::new(999, 1))
+            );
+            assert_eq!(
+                BasicVersion::parse("1.999"),
+                Some(BasicVersion::new(1, 999))
+            );
+            assert_eq!(BasicVersion::parse("01.02"), Some(BasicVersion::new(1, 2)));
+            assert_eq!(BasicVersion::parse("1.02"), Some(BasicVersion::new(1, 2)));
+            assert_eq!(BasicVersion::parse("01.2"), Some(BasicVersion::new(1, 2)));
+        }
+
+        #[test]
+        fn test_parse_invalid_versions() {
+            // No dot
+            assert_eq!(BasicVersion::parse("1"), None);
+            assert_eq!(BasicVersion::parse("123"), None);
+
+            // Multiple dots
+            assert_eq!(BasicVersion::parse("1.2.3"), None);
+            assert_eq!(BasicVersion::parse("1.2.3.4"), None);
+
+            // Empty string
+            assert_eq!(BasicVersion::parse(""), None);
+
+            // Only dot
+            assert_eq!(BasicVersion::parse("."), None);
+
+            // Missing major or minor
+            assert_eq!(BasicVersion::parse(".2"), None);
+            assert_eq!(BasicVersion::parse("1."), None);
+
+            // Non-numeric values
+            assert_eq!(BasicVersion::parse("a.b"), None);
+            assert_eq!(BasicVersion::parse("1.b"), None);
+            assert_eq!(BasicVersion::parse("a.2"), None);
+            assert_eq!(BasicVersion::parse("1.2a"), None);
+            assert_eq!(BasicVersion::parse("1a.2"), None);
+
+            // Negative numbers
+            assert_eq!(BasicVersion::parse("-1.2"), None);
+            assert_eq!(BasicVersion::parse("1.-2"), None);
+            assert_eq!(BasicVersion::parse("-1.-2"), None);
+
+            // Whitespace
+            assert_eq!(BasicVersion::parse(" 1.2"), None);
+            assert_eq!(BasicVersion::parse("1.2 "), None);
+            assert_eq!(BasicVersion::parse("1 .2"), None);
+            assert_eq!(BasicVersion::parse("1. 2"), None);
```

Comment:
May be add a test case for overflow as well e.g. for major or minor strings having values bigger than `u32::MAX`